route/build/src/datetime_data.f90

```diff
@@ -147,47 +160,66 @@ SUBROUTINE sub_assign(this, that)
     this%ih   = that%ih
     this%imin = that%imin
     this%dsec = that%dsec
+    this%calendar = that%calendar
   END SUBROUTINE sub_assign
 
 
-  integer(i4b) FUNCTION fn_get_year(this)
+  pure elemental integer(i4b) FUNCTION fn_get_year(this)
     implicit none
     class(datetime), intent(in) :: this
     fn_get_year = this%iy
   END FUNCTION fn_get_year
 
-  integer(i4b) FUNCTION fn_get_month(this)
+  pure elemental integer(i4b) FUNCTION fn_get_month(this)
     implicit none
     class(datetime), intent(in) :: this
     fn_get_month = this%im
   END FUNCTION fn_get_month
 
-  integer(i4b) FUNCTION fn_get_day(this)
+  pure elemental integer(i4b) FUNCTION fn_get_day(this)
     implicit none
     class(datetime), intent(in) :: this
     fn_get_day = this%id
   END FUNCTION fn_get_day
 
-  integer(i4b) FUNCTION fn_get_hour(this)
+  pure elemental integer(i4b) FUNCTION fn_get_hour(this)
     implicit none
     class(datetime), intent(in) :: this
     fn_get_hour = this%ih
   END FUNCTION fn_get_hour
 
-  integer(i4b) FUNCTION fn_get_min(this)
+  pure elemental integer(i4b) FUNCTION fn_get_min(this)
     implicit none
     class(datetime), intent(in) :: this
     fn_get_min = this%imin
   END FUNCTION fn_get_min
 
-  real(dp) FUNCTION fn_get_sec(this)
+  pure elemental real(dp) FUNCTION fn_get_sec(this)
     implicit none
     class(datetime), intent(in) :: this
     fn_get_sec = this%dsec
   END FUNCTION fn_get_sec
 
+  pure elemental character(20) FUNCTION fn_get_calendar(this)
+    implicit none
+    class(datetime), intent(in) :: this
+    fn_get_calendar = this%calendar
+  END FUNCTION fn_get_calendar
+
+
+  integer(i4b) FUNCTION fn_dayofyear(this)
+    ! Returns the integer day of the year (ordinal date).
+    implicit none
+    class(datetime), intent(in) :: this
+    type(datetime)              :: jan1datetime
+
+    jan1datetime = datetime(this%iy, 1, 1, 0, 0, 0.0, calendar=this%calendar)
+    fn_dayofyear = int((this - jan1datetime)/86400._dp) + 1
```

Comment:
Better to use a parameter for 86400. rather than a magic number

Reply:
replaced 86400 with secprday